Mathlib/Data/Nat/Digits.lean

```diff
@@ -549,92 +538,96 @@ theorem ofDigits_zmod (b : ℤ) (k : ℕ) (L : List ℕ) : ofDigits b L % k = of
 theorem ofDigits_neg_one :
     ∀ L : List ℕ, ofDigits (-1 : ℤ) L = (L.map fun n : ℕ => (n : ℤ)).alternatingSum
   | [] => rfl
-  | [n] => by simp [of_digits, List.alternatingSum]
-  | a::b::t =>
+  | [n] => by simp [ofDigits, List.alternatingSum]
+  | a :: b :: t =>
     by
-    simp only [of_digits, List.alternatingSum, List.map_cons, of_digits_neg_one t]
+    simp only [ofDigits, List.alternatingSum, List.map_cons, ofDigits_neg_one t]
     ring
 #align nat.of_digits_neg_one Nat.ofDigits_neg_one
 
 theorem modEq_eleven_digits_sum (n : ℕ) :
     n ≡ ((digits 10 n).map fun n : ℕ => (n : ℤ)).alternatingSum [ZMOD 11] := by
-  have t := zmodeq_of_digits_digits 11 10 (-1 : ℤ) (by unfold Int.ModEq <;> norm_num) n
-  rwa [of_digits_neg_one] at t
+  have t := zmodeq_ofDigits_digits 11 10 (-1 : ℤ) (by unfold Int.ModEq; norm_num) n
+  rwa [ofDigits_neg_one] at t
 #align nat.modeq_eleven_digits_sum Nat.modEq_eleven_digits_sum
 
 /-! ## Divisibility  -/
 
 
 theorem dvd_iff_dvd_digits_sum (b b' : ℕ) (h : b' % b = 1) (n : ℕ) :
-    b ∣ n ↔ b ∣ (digits b' n).Sum := by
-  rw [← of_digits_one]
-  conv_lhs => rw [← of_digits_digits b' n]
-  rw [Nat.dvd_iff_mod_eq_zero, Nat.dvd_iff_mod_eq_zero, of_digits_mod, h]
+    b ∣ n ↔ b ∣ (digits b' n).sum := by
+  rw [← ofDigits_one]
+  conv_lhs => rw [← ofDigits_digits b' n]
+  rw [Nat.dvd_iff_mod_eq_zero, Nat.dvd_iff_mod_eq_zero, ofDigits_mod, h]
 #align nat.dvd_iff_dvd_digits_sum Nat.dvd_iff_dvd_digits_sum
 
 /-- **Divisibility by 3 Rule** -/
-theorem three_dvd_iff (n : ℕ) : 3 ∣ n ↔ 3 ∣ (digits 10 n).Sum :=
+theorem three_dvd_iff (n : ℕ) : 3 ∣ n ↔ 3 ∣ (digits 10 n).sum :=
   dvd_iff_dvd_digits_sum 3 10 (by norm_num) n
 #align nat.three_dvd_iff Nat.three_dvd_iff
 
-theorem nine_dvd_iff (n : ℕ) : 9 ∣ n ↔ 9 ∣ (digits 10 n).Sum :=
+theorem nine_dvd_iff (n : ℕ) : 9 ∣ n ↔ 9 ∣ (digits 10 n).sum :=
   dvd_iff_dvd_digits_sum 9 10 (by norm_num) n
 #align nat.nine_dvd_iff Nat.nine_dvd_iff
 
 theorem dvd_iff_dvd_ofDigits (b b' : ℕ) (c : ℤ) (h : (b : ℤ) ∣ (b' : ℤ) - c) (n : ℕ) :
     b ∣ n ↔ (b : ℤ) ∣ ofDigits c (digits b' n) := by
   rw [← Int.coe_nat_dvd]
   exact
-    dvd_iff_dvd_of_dvd_sub (zmodeq_of_digits_digits b b' c (Int.modEq_iff_dvd.2 h).symm _).symm.Dvd
+    dvd_iff_dvd_of_dvd_sub (zmodeq_ofDigits_digits b b' c (Int.modEq_iff_dvd.2 h).symm _).symm.dvd
 #align nat.dvd_iff_dvd_of_digits Nat.dvd_iff_dvd_ofDigits
 
 theorem eleven_dvd_iff :
     11 ∣ n ↔ (11 : ℤ) ∣ ((digits 10 n).map fun n : ℕ => (n : ℤ)).alternatingSum := by
-  have t := dvd_iff_dvd_of_digits 11 10 (-1 : ℤ) (by norm_num) n
-  rw [of_digits_neg_one] at t
+  have t := dvd_iff_dvd_ofDigits 11 10 (-1 : ℤ) (by norm_num) n
+  rw [ofDigits_neg_one] at t
   exact t
 #align nat.eleven_dvd_iff Nat.eleven_dvd_iff
 
+-- Porting note: Is this proved somewhere in mathlib?
+theorem pow_zero_ {n : ℤ} : n ^ 0 = 1 := by simp
+theorem pow_succ_ {n : ℤ} {p : ℕ} : n ^ (succ p) = n * (n ^ p) := by
```

Comment:
They should simply be `pow_zero` and `pow_succ`. Would you need an extra import for those?
It looks like `pow_succ` was available here in mathlib3.

Reply:
Hm. Looks like the problem is that [`pow_succ`](https://github.com/leanprover-community/mathlib4/blob/a0a7b109e62bfa3d0f3355091cd112928f398d67/Mathlib/Algebra/Group/Defs.lean#L630) is getting shadowed by the more specialized [`Nat.pow_succ`](https://github.com/leanprover/lean4/blob/3f6c5f17db30c2a2dcaca1b9e9faec4e2cf04b95/src/Init/Data/Nat/Basic.lean#L473). It works if I fully qualify the `pow_succ` here as `_root_.pow_succ`.